MapboxNavigation/Cache.swift

```diff
@@ -109,22 +109,22 @@ internal class FileCache {
         }
     }
 
-    private func cachePathWithKey(_ key: String) -> String {
+    func cachePathWithKey(_ key: String) -> String {
         let cacheKey = cacheKeyForKey(key)
         return cacheURLWithKey(cacheKey).absoluteString
     }
 
-    private func cacheURLWithKey(_ key: String) -> URL {
+    func cacheURLWithKey(_ key: String) -> URL {
         let cacheKey = cacheKeyForKey(key)
         return diskCacheURL.appendingPathComponent(cacheKey)
     }
 
-    private func cacheKeyForKey(_ key: String) -> String {
+    func cacheKeyForKey(_ key: String) -> String {
         if let keyAsURL = URL(string: key) {
-            return String.init(keyAsURL.lastPathComponent.hashValue)
+            return MD5(keyAsURL.lastPathComponent)
```

Comment:
I guess I'm surprised to see us still keying off of `lastPathComponent` -- I thought this was part of what was biting us. (?)

Reply:
Not that I am aware of. It really only came down to two different strings having the same hashValue: https://github.com/mapbox/mapbox-navigation-ios/issues/1391#issuecomment-387490015

Reply:
I'm happy to remove this check though. I think creating an md5 of the `key` here would be okay.